mvpa2/algorithms/hyperalignment.py

```diff
@@ -487,27 +498,48 @@ def _level3(self, datasets):
 
         # key different from level-2; the common space is uniform
         #temp_commonspace = commonspace
-
-        residuals = None
-        if self.ca['residual_errors'].enabled:
-            residuals = np.zeros((1, len(datasets)))
-            self.ca.residual_errors = Dataset(samples=residuals)
+        # Checking for joblib, if not, set nproc to 1
+        if params.nproc > 1:
+            from mvpa2.base import externals, warning
+            if not externals.exists('joblib'):
+                warning("Setting nproc > 1 requires joblib package, which "
+                        "does not seem to exist. Setting nproc to 1.")
+                params.nproc = 1
 
         # start from original input datasets again
-        for i, (m, ds_new) in enumerate(zip(mappers, datasets)):
+        if params.nproc == 1:
+            residuals = []
+            for i, (m, ds_new) in enumerate(zip(mappers, datasets)):
+                if __debug__:
+                    debug('HPAL_', "Level 3: ds #%i" % i)
+                m, residual = get_trained_mapper(ds_new, self.commonspace, m,
+                                                 self.ca['residual_errors'].enabled)
+                if self.ca['residual_errors'].enabled:
+                    residuals.append(residual)
+        else:
             if __debug__:
-                debug('HPAL_', "Level 3: ds #%i" % i)
+                debug('HPAL_', "Level 3: Using joblib with %d jobs" % params.nproc)
+            verbose_level_parallel = 20 \
+                if (__debug__ and 'HPAL' in debug.active) else 0
+            from joblib import Parallel, delayed
+            import sys
+            # joblib's 'multiprocessing' backend has known issues of failure on OSX
+            # Tested with MacOS 10.12.13, python 2.7.13, joblib v0.10.3
+            if params.joblib_backend is None:
+                params.joblib_backend = 'threading' if sys.platform == 'darwin' \
+                                        else'multiprocessing'
```

Comment:
missing space before `'multiprocessing'` -- and tests didn't scream "SyntaxError" ? ;-)